### PR TITLE
Fix SonarQube issues in image.cpp

### PR DESCRIPTION
**Pull request by Districh-ru**

### CODE REVIEW

File: src/engine/image.cpp [hunk 1]
```diff
@@ -597,10 +597,10 @@ namespace fheroes2
     Sprite & Sprite::operator=( Sprite && sprite ) noexcept
     {
         if ( this != &sprite ) {
-            Image::operator=( std::move( sprite ) );
-
             std::swap( _x, sprite._x );
             std::swap( _y, sprite._y );
+
+            Image::operator=( std::move( sprite ) );
```
Comment (oleg-derevenetz) [inline src/engine/image.cpp:603]: Does this change need to convince SonarQube of the correctness of the logic, or is it just a change of some kind of aesthetic nature?

Comment (ihhub) [reply src/engine/image.cpp:603]: I suggest to revert this change if it makes not difference in SonarQube analysis results.

**Review by ihhub — APPROVED** [review]

File: src/engine/image.cpp (hunk 1 above)
Comment (Districh-ru) [reply src/engine/image.cpp:603]: Hi, @oleg-derevenetz  and @ihhub, this change is done to fix Clang-Tidy issue: `'sprite' used after it was moved [bugprone-use-after-move]`

**Comment by ihhub** [comment]
@Districh-ru , thank you very much for these fixes!

**Pull request merged**